src/Css/CssDocument.php

```diff
@@ -37,10 +37,22 @@ class CssDocument
 
     /**
      * @param string $css
+     * @param bool $debug
+     *        If this is `true`, an exception will be thrown if invalid CSS is encountered.
+     *        Otherwise the parser will try to do the best it can.
      */
-    public function __construct(string $css)
+    public function __construct(string $css, bool $debug)
     {
-        $this->sabberwormCssDocument = (new CssParser($css))->parse();
+        // CSS Parser currently throws exception with nested at-rules (like `@media`) in strict parsing mode
+        // @see https://github.com/sabberworm/PHP-CSS-Parser/issues/127
+        $parserSettings = \Sabberworm\CSS\Settings::create()->withLenientParsing(
+            !$debug ||
+            \preg_match('/@(?:media|supports|(?:-webkit-|-moz-|-ms-|-o-)?+(keyframes|document))\\b/', $css) === 1
```

Comment:
Please let's move this `preg_match` to a local variable with a name that communicates its meaning.

Reply:
Also added a `use` for the settings class.